components/studio/bin/hab-studio.sh

```diff
@@ -1160,6 +1097,33 @@ set_libexec_path() {
   return 0
 }
 
+defaults_path() {
+    echo "$($bb dirname "${libexec_path}")/defaults"
+}
+
+# If `file_path` is not present in the studio, copy in a minimal
+# default version from the studio package's `defaults` directory.
+copy_minimal_default_file_if_not_present() {
+    local file_path="${1}"
+    if [ ! -f "${HAB_STUDIO_ROOT}${file_path}" ]; then
+        copy_minimal_default_file "${file_path}"
+    fi
+}
+
+copy_minimal_default_file() {
+    local file_path="${1}"
+    if [ -n "$VERBOSE" ]; then
+        echo "> Creating minimal ${file_path}"
+    fi
+    if [ ! -f "$(defaults_path)${file_path}" ]; then
+        echo
+        echo "Tried to copy default file for '${file_path}', but could not find one! Please report this error."
+        echo
+    else
+        $bb cp "$(defaults_path)${file_path}" "${HAB_STUDIO_ROOT}${file_path}"
+    fi
```

Comment:
Personally, I'd turn the if statement into a positive conditional, as I think it makes it a bit easier to read, e.g.

```shell
if [ -f "$(defaults_path)${file_path}" ]; then
  $bb cp "$(defaults_path)${file_path}" "${HAB_STUDIO_ROOT}${file_path}"
else
  echo
  echo "Tried to copy default file for '${file_path}', but could not find one! Please report this error."
  echo
fi
```

Reply:
Excellent point; will change.

Reply:
Fixed with https://github.com/habitat-sh/habitat/pull/6319/commits/31e2257f7c8b27668aeca7720cc53d92fa2f5022